netket/operator/_operator_list.py

```diff
@@ -0,0 +1,54 @@
+from typing import Union
+import numpy as np
+
+
+class OperatorList:
+    """
+    Data class to be used internally to handle the matrix representation of
+    different operators. This class helps in storing only the necessary
+    matrices in memory. The instances of this class behave like a list,
+    but they only store unrepeated operators.
+    """
+
+    def __init__(self):
+        self._operators = {}
+        self._index2operatorkeys = []
+
+    def __getitem__(self, key: Union[int, slice]) -> np.ndarray:
+        if isinstance(key, slice):
+            length = len(self._index2operatorkeys)
+            return [
+                self._operators[self._index2operatorkeys[k]]
+                for k in range(*key.indices(length))
+            ]
+        return self._operators[self._index2operatorkeys[key]]
+
+    def __setitem__(self, key: int, operator: np.ndarray) -> None:
+        value_hash = hash(operator.tostring())
```

Comment:
`tostring` is deprecated in favor of `tobytes` (which is equivalent, see [here](https://numpy.org/doc/stable/reference/generated/numpy.ndarray.tostring.html)).
```suggestion
        value_hash = hash(operator.tobytes())
```